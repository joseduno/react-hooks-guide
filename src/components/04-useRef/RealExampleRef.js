import React, { useState } from 'react'
// import useFetch from '../../hooks/useFetch'
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks'

const RealExampleRef = () => {

    const [show, setShow] = useState(false)

    return (
        <div className='container mt-5'>
            <h1>RealExampleRef</h1>
            <button className='btn btn-primary mt-1' onClick={()=>{setShow(!show)}}>Show/Hide</button>
            {show && <MultipleCustomHooks />}
            <hr />
        </div>
    )
}

export default RealExampleRef
