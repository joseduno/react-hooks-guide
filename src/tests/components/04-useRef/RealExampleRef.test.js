import { shallow } from "enzyme";
import RealExampleRef from "../../../components/04-useRef/RealExampleRef";

describe('Pruebas en el componente <RealExampleRef />', () => {

    const wrapper = shallow(<RealExampleRef />)

    test('debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false)
    })

    test('debe mostrarse el componente hijo <MultipleCustomHooks />', () => {
        wrapper.find('button').simulate('click')
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true)
    })
    
    
})
