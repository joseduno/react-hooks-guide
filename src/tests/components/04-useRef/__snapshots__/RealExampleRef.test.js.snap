// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <RealExampleRef /> debe mostrar el componente correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    RealExampleRef
  </h1>
  <button
    className="btn btn-primary mt-1"
    onClick={[Function]}
  >
    Show/Hide
  </button>
  <hr />
</div>
`;
